### PR TITLE
Use non-deprecated branch protection policy fields

### PR DESCRIPTION
**Pull request by fejta**
/assign @stevekuznetsov @cjwagner  

Also minimal sanity check on concrete config.

### CODE REVIEW

File: prow/config/config_test.go [hunk 1]
```diff
@@ -1156,3 +1156,34 @@ func TestDecorationDefaulting(t *testing.T) {
...
+	}
+	for _, tc := range cases {
+		policy, err := c.GetBranchProtection(tc.org, tc.repo, tc.branch)
+		if err != nil {
+			t.Errorf("%s/%s=%s: unexpected error: %v", tc.org, tc.repo, tc.branch, err)
+		}
+		if policy.Protect == nil || !*policy.Protect {
+			t.Errorf("%s/%s=%s: unprotected: %v", tc.org, tc.repo, tc.branch, policy.Protect)
+		}
+		if !sets.NewString(policy.RequiredStatusChecks.Contexts...).Has("cla/linuxfoundation") {
```
Comment (BenTheElder) [inline prow/config/config_test.go]: it seems a little off to couple tests to the kubernetes/* prow config. /shrug

Comment (BenTheElder) [reply prow/config/config_test.go]: I mean we have plenty of that now, but it seems like a direction we should move away from..?

Comment (fejta) [reply prow/config/config_test.go]: I agree we should separate kubernetes config testing from prow/config package testing... But I think config_test.go already conflates this, and I would rather pull these apart in its own PR

Comment (BenTheElder) [reply prow/config/config_test.go]: SGTM

**Pull request merged**